src/contentscript/index.ts

```diff
@@ -1,19 +1,44 @@
 
 import { injectAllScripts } from './inject';
 import { IExtensionAPIMessage, IRPCCallRequest, IRPCCallResponse, ICurrentAccount } from '../types';
-import { TARGET_NAME, API_TYPE, MESSAGE_TYPE, RPC_METHOD } from '../constants';
+import { TARGET_NAME, API_TYPE, MESSAGE_TYPE, RPC_METHOD, PORT_NAME } from '../constants';
 import { isMessageNotValid } from '../utils';
 import { postWindowMessage } from '../utils/messenger';
 
+let port: any;
+
 // Inject scripts
 injectAllScripts();
 
 // Add message listeners
 window.addEventListener('message', handleInPageMessage, false);
+// TODO remove the line below after transitioning rpc to long lived connection
 chrome.runtime.onMessage.addListener(handleBackgroundScriptMessage);
+// Dapp developer triggers this event to set up window.qrypto
+window.addEventListener('message', setupLongLivedConnectionToBP, false);
```

Comment:
I think you mean single listener between
1)chrome.runtime.onMessage.addListener(handleBackgroundScriptMessage);
and 2) window.addEventListener('message', setupLongLivedConnectionToBP, false);
right?

I tested it, we can't do it because these are different types of messages (chrome.runtime message and window message). Messages sent via chrome.runtime from the bg aren't heard by window.addEventListener.  And I believe you  need to use chrome.runtime to reach the bg page. 
I also dont think they are actually the same type of message, one is a MessageEvent the other is a window event even though they are both called "message"

Reply:
Oops had tunnel vision and didnt see the prefix.